imap/mailbox.c

```diff
@@ -2105,7 +2105,7 @@ static int mailbox_buf_to_index_record(const char *buf, int version,
     }
 
     cache_offset_field = ntohl(*((bit32 *)(buf+OFFSET_CACHE_OFFSET)));
-    record->internaldate = ntohll(*((bit64 *)(buf+OFFSET_INTERNALDATE)));
+    record->internaldate.tv_sec  = ntohll(*((bit64 *)(buf+OFFSET_INTERNALDATE)));
```

Comment:
Is this portable on a 32 bit system where long long will be 64 bits but time_t could be 32? Do we care?

Reply:
We would have the same problem before this change with record->internaldate which was also time_t.

Do you have any suggestions?

Reply:
Hmm fair point. I suppose on such systems internaldate will be 32bits anyway so maybe this is fine? Alright ignore this

Reply:
Actually, this conversion isn't even in the current code anymore.  We do assign tv_sec the result of  uint64 / 1000000000, but we won't have to worry about that until 2038.

Reply:
FWIW: We already don't support weird platforms where time_t is not the system's native word size (e.g. some mobile device architectures have 32-bit cpus but a 64-bit time_t), and we barely support platforms where both are 32-bit.  See extensive discussion in #4876 and related PRs etc.

Though as far as I understand, any OS we support, on any hardware that can provide a 64-bit integer, will use that for time_t.  So we might need to worry about 'long' or 'int' being different from time_t (e.g. those 32-bit mobile devices), but not 'long long'.  At least until 128-bit integer types become common...